src/function/algebra/simplify.js

```diff
@@ -495,8 +495,17 @@ export const createSimplify = /* #__PURE__ */ factory(name, dependencies, (
       assuming: { multiply: { associative: true } }
     },
 
-    { l: 'n1/(-n2)', r: '-n1/n2' }
+    { l: 'n1/(-n2)', r: '-n1/n2' },
 
+    {
+      s: '(c1*v+c2)/c3 -> ((c1*v)/c3)+c2/c3',
+      assuming: { multiply: { associative: true } }
+    },
+
+    {
+      s: '(-(c1*v)-c2)/c3 -> (-c1/c3)*v+-(c2/c3)',
+      assuming: { multiply: { associative: true } }
+    }
   ]
```

Comment:
Thanks for finding these expressions that help with #2594. I am slightly confused by them -- there are two sign variants, but the sign is changed in two places. Are the other two possibilities where only one sign is changed instead of both needed? Why not? (Also I am unclear on the need for associativity in this identity; it seems to use only the distributive property, but I think we always assume that, I don't think there is a separate assumption for that.) Conversely, is there any way that these expressions can be merged into one -- that would help slow the ongoing growth of complication in simplify. I know there is a step where signs are merged into constants -- could this simplification possibly work with a single sign variant at a differentlocation? As another idea, would casting the rule as
```
(c1*v +c2)/c3 -> ((c1/c3)*v + c2/c3)
```
(this would now need a commutative assumption) perform any better, as it coalesces constants? Just some thoughts, I haven't tried any alternatives.  Thanks for considering.

Reply:
Ah, sorry, I hadn't focused entirely on Jos's feedback, which is excellent. Did you try his suggestion `(n1+c2)/c3 -> n1/c3 + c2/c3`? That might work well, especially in a place where all signs have been absorbed into constants so that there isn't any such thing as `n1-c2` at that point. That might avoid the need for a function as Jos suggests because perhaps all of the "other stuff" that could be in the numerator that Jos mentions could be absorbed into the n1 term. Or possibly it could help to write it as `(c2+n1)/c3 -> c2/c3 + n1/c3` because the simplifier prefers to match on the beginnings of terms? It is true and unfortunate that its operation is a bit finicky...  Thanks for following up on these ideas if you can, and I was just about to add the comment about tests when I saw Jos already had.